execute/table.go

```diff
@@ -1323,9 +1323,33 @@ func (t *ColListTable) Do(f func(flux.ColReader) error) error {
 	if !atomic.CompareAndSwapInt32(&t.used, 0, 1) {
 		return errors.New(codes.Internal, "table already read")
 	}
+
 	var err error
 	if t.nrows > 0 {
-		err = f(t)
+		buffer := &arrow.TableBuffer{
+			GroupKey: t.key,
+			Columns:  t.colMeta,
+			Values:   make([]array.Array, len(t.cols)),
+		}
+		for i, col := range t.cols {
+			switch col := col.(type) {
+			case *boolColumn:
+				buffer.Values[i] = col.data
+			case *intColumn:
+				buffer.Values[i] = col.data
+			case *uintColumn:
+				buffer.Values[i] = col.data
+			case *floatColumn:
+				buffer.Values[i] = col.data
+			case *stringColumn:
+				buffer.Values[i] = col.data
+			case *timeColumn:
+				buffer.Values[i] = col.data
+			default:
+				return errors.Newf(codes.Internal, "unknown column type: %T", col)
+			}
+		}
+		err = f(buffer)
 		t.Release()
```

Comment:
My reading of this is we're now copying the data out of the incoming table, into a fresh buffer, passing that new buffer to `f()`, and releasing the incoming table.

For my own edification - what's the significance of this? Is it because otherwise we don't know when `t` should be released otherwise?

Reply:
There was some complicated logic happening with the old version. It implemented both `flux.Table` and `flux.ColReader`, but it caused it act weird if the column reader was retained after the table was read. So I replaced it with just our standard `flux.ColReader` implementation so there weren't any weird data flows.